src/stan/io/reader.hpp

```diff
@@ -2,9 +2,9 @@
 #define STAN__IO__READER_HPP
 
 #include <boost/throw_exception.hpp>
+#include <stan/math/prim/scal.hpp>
```

Comment:
All these order changes are due to the following problem. Perhaps there is a better way to resolve it?

```
In file included from src/stan/math/prim/arr.hpp:7:0,
                 from src/stan/io/reader.hpp:5,
                 from src/test/unit/io/reader_test.cpp:1:
src/stan/math/prim/arr/meta/is_vector_like.hpp:7:10: error: ‘is_vector_like’ is not a class template
```